Include/object.h

```diff
@@ -102,26 +102,62 @@ typedef struct _typeobject PyTypeObject;
  * by hand.  Similarly every pointer to a variable-size Python object can,
  * in addition, be cast to PyVarObject*.
  */
-typedef struct _object {
+
+typedef struct _object      PyObject;
+typedef struct _varobject   PyVarObject;
+
+// TODO: Send "_object" and "_varobject" to CPython internal only.
+struct _object {
     _PyObject_HEAD_EXTRA
     Py_ssize_t ob_refcnt;
     PyTypeObject *ob_type;
-} PyObject;
+};
+
+struct _varobject {
+    PyObject ob_base;
+    Py_ssize_t ob_size; /* Number of items in variable part */
+};
 
 /* Cast argument to PyObject* type. */
 #define _PyObject_CAST(op) ((PyObject*)(op))
 #define _PyObject_CAST_CONST(op) ((const PyObject*)(op))
 
-typedef struct {
-    PyObject ob_base;
-    Py_ssize_t ob_size; /* Number of items in variable part */
-} PyVarObject;
-
 /* Cast argument to PyVarObject* type. */
 #define _PyVarObject_CAST(op) ((PyVarObject*)(op))
 #define _PyVarObject_CAST_CONST(op) ((const PyVarObject*)(op))
 
+Py_SLIB_LOCAL(Py_ssize_t) PyObject_GetRefCount(const PyObject *ob);
+Py_SLIB_LOCAL(PyTypeObject *) PyObject_GetType(const PyObject *ob);
+
+Py_SLIB_LOCAL(void) PyObject_SetRefCount(PyObject *ob, const Py_ssize_t refcnt);
+Py_SLIB_LOCAL(void) PyObject_SetType(PyObject *ob, const PyTypeObject *type);
```

Comment:
Python 3.9 already have Py_SET_REFCNT() and Py_SET_TYPE() functions. What is the advantage of adding new functions?

Reply:
These aren't essentially "new functions", they are bridges into CPython's internals WITHOUT leaking implementation details.

As mentioned before, the existing functions WILL FAIL when PyObject goes full opaque.